src/combinatorics/rado/auxiliary.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+
+import data.finset.card
+import data.fintype.basic
+
+/-!
+# Some auxiliary lemmas on finsets etc.
+
+These should probably go into `data.finset.basic`, `data.finset.image` or `data.fintype.basic`.
+-/
+
+namespace finset
+
+/-!
+Define versions of `eq_on`, `inj_on` and `restrict` for `finset`s.
+
+This is mostly for convenience so that we can use dot notation for these on `finset`s.
+-/
+
+section finset_versions
+
+variables {α : Type*} {β : Type*}
+
+/-- The restriction of `f` to a `finset` `s` -/
+def restrict (s : finset α) (f : α →  β) (a : ↥s) : β := f a
+
+/-- The statement that two functions `f` and `g` agree on a `finset` `s` -/
+def eq_on (f g : α → β) (s : finset α) : Prop := (s : set α).eq_on f g
+
+/-- The statement that a function `f` is injective on a `finset` `s` -/
+def inj_on (f : α → β) (s : finset α) : Prop := (s : set α).inj_on f
+
+-- We don't need the next two, but include them for completeness
+
+/-- The statement that a function `f` is surjective from a `finset` `s` to a `finset` `t` -/
+def surj_on (f : α → β) (s : finset α) (t : finset β) : Prop := (s : set α).surj_on f t
+
+/-- The statement that a function `f` is bijective from a `finset` `s` to a `finset` `t` -/
+def bij_on (f : α → β) (s : finset α) (t : finset β) : Prop := (s : set α).bij_on f t
+
+end finset_versions
```

Comment:
I doubt we want those. Maybe @kmill can set us up some magic that makes dot notation on finset resolve to the corresponding `set` one?

Reply:
My experience is that dot notation does not work well with casts...

Reply:
Phrased another way; I think there are many places in mathlib that put up with the absence of these definitions; I think it would be best to continue to do so, and leave deciding whether to add them till mathlib4.

Reply:
There is [finset.image](https://leanprover-community.github.io/mathlib_docs/data/finset/image.html#finset.image), which is similar in spirit, but probably owes its existence to the fact that one wants a `finset` as the image of a `finset`, which is maybe not straightforward via [set.image](https://leanprover-community.github.io/mathlib_docs/data/set/image.html#set.image).

Reply:
Anyway, what you seem to suggest is: remove these for now and live with the clumsier notation?

Reply:
OK.